### ISSUE TITLE
Rule suggestion: no thenables in conditions

### ISSUE THREAD

**Issue by Sammons** [palantir/tslint#1496]
A common mistake I've been making when using await/async functionality in the latest typescript versions is 

```ts
let someEntity = someRepo.oneWhere({ id: id })// returns bluebird<{ id: id, otherprops }>;
if (someEntity) {
 // assume exists
}
```
when the correct code in our codebase is
```ts
let someEntity = await someRepo.oneWhere({ id: id })// returns bluebird<{ id: id, otherprops }>;
if (someEntity) {
 // assume exists
}
```

forgetting the `await` is really easy, and not so easy to spot. Would a pull request with this rule stand a chance of making it in?

**Comment by adidahiya** [comment]
I think this lint pattern is more broadly captured by https://github.com/palantir/tslint/issues/253 -- would you prefer that general rule or a more specific one for thenables / promises / etc?

**Comment by Sammons** [comment]
I didn't see that one -- I think it would be best to allow booleans only and whitelist types. So maybe slightly more generic?

**Comment by JoshuaKGoldberg** [comment]
Related to https://github.com/Microsoft/tslint-microsoft-contrib/issues/174 from @sethob

**Comment by Sammons** [comment]
I'm closing this since I think either https://github.com/Microsoft/tslint-microsoft-contrib/issues/174 or #253 will cover what I need. Thanks guys!
